flink-yarn/src/main/java/org/apache/flink/yarn/Utils.java

```diff
@@ -351,16 +351,16 @@ static ContainerLaunchContext createTaskExecutorContext(
 		require(yarnClientUsername != null, "Environment variable %s not set", YarnConfigKeys.ENV_HADOOP_USER_NAME);
 
 		final String remoteKeytabPath = env.get(YarnConfigKeys.KEYTAB_PATH);
-		log.info("TM:remote keytab path obtained {}", remoteKeytabPath);
-
 		final String remoteKeytabPrincipal = env.get(YarnConfigKeys.KEYTAB_PRINCIPAL);
-		log.info("TM:remote keytab principal obtained {}", remoteKeytabPrincipal);
-
 		final String remoteYarnConfPath = env.get(YarnConfigKeys.ENV_YARN_SITE_XML_PATH);
-		log.info("TM:remote yarn conf path obtained {}", remoteYarnConfPath);
-
 		final String remoteKrb5Path = env.get(YarnConfigKeys.ENV_KRB5_PATH);
-		log.info("TM:remote krb5 path obtained {}", remoteKrb5Path);
+
+		if (log.isDebugEnabled()) {
```

Comment:
the {{isDebugEnabled}} is typically used to avoid unnecessary String concatination.
but checking for debugging enabled is not necessary here.

Reply:
@yew1eb , In a way, I agree with you. If there is just one `log.debug('xxx')` and no string concatination, the `if (log.isDebugEnabled())` is not necessary. But this case, there are four `log.debug()` call , and each of them would do the same judgement inside the `debug` method. So wrapping a outer judgement is performance reason.

Reply:
Sorry, I forgot to tell you that the `log.debug()` method internally calls `islogEnabled()`.
You can see the `org.slf4j.Logger`'s log4j adapter class:
```
  //org.slf4j.impl.Log4jLoggerAdapter.java
...
  public void debug(String format, Object arg) {
    if (logger.isDebugEnabled()) {
      FormattingTuple ft = MessageFormatter.format(format, arg);
      logger.log(FQCN, Level.DEBUG, ft.getMessage(), ft.getThrowable());
    }
  }
...
```

Reply:
I know as I said in the previous comment `and each of them would do the same judgement inside the debug method.`. Here I do a judgement to avoid that if the log level upper then DEBUG, it could just jump these code.